src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -32,15 +44,16 @@ export class ItemChildrenEditComponent implements OnChanges {
   @Input() itemData?: ItemData;
 
   state: 'loading' | 'error' | 'ready' = 'ready';
-  data: ChildData[] = [];
-  selectedRows: ChildData[] = [];
+  data: (ChildData&ChildDataAdditions)[] = [];
```

Comment:
If you repeat `ChildData&ChildDataAdditions` (4x I think), just define a type `type ... = ChildData&ChildDataAdditions`.
BTW, why everything is not directly in `ChildData`?

Reply:
Fixed